ui/desktop/electron-app/src/index.js

```diff
@@ -284,18 +285,23 @@ app.on('ready', async () => {
 });
 
 /**
- * Prompt for closing spawned processes
+ * Prompt use to confirm action before closing spawned processes
+ * This prompt lives in the ember app using Hds::Modal
  */
-app.on('before-quit', (event) => {
+app.on('before-quit', async (event) => {
+  if (app.signoutInProgress) {
+    event.preventDefault();
+    return;
+  }
+
   if (sessionManager.hasRunningSessions) {
-    const dialogOpts = {
-      type: 'question',
-      buttons: ['Close', 'Cancel'],
-      detail: 'Close sessions before quitting?',
-    };
-
-    const buttonId = dialog.showMessageBoxSync(null, dialogOpts);
-    buttonId === 0 ? sessionManager.stopAll() : event.preventDefault();
+    try {
+      await mainWindow.webContents.send('inform-ember-of-app-closure');
+      event.preventDefault();
+      return;
+    } catch (e) {
+      console.log(`Exception: ${e}`);
+    }
```

Comment:
I don't think you need the `signoutInProgress` here anymore. All this function needs to do is to check if it has running sessions and if it does, send the signal to let the UI handle it and prevent the app from quitting since the UI will send the signal to quit now.